com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -485,6 +586,399 @@ private void OnValidate()
+                        InvokeOnReadyCallback();
+                    }
+                    else
+                    {
+                        Debug.LogError($"Server is shutting down due to network transport start failure of {NetworkConfig.NetworkTransport.GetType().Name}!");
+                        InvokeOnStartupFailedCallback(StartupFailureReason.BindFailed);
+                        ShutdownInternal();
+                        return false;
+                    }
+                }
+
+                if (IsHost)
+                {
+                    if (NetworkConfig.ConnectionApproval)
+                    {
+                        InvokeConnectionApproval(NetworkConfig.ConnectionData, ServerClientId,
+                            (createPlayerObject, playerPrefabHash, approved, position, rotation) =>
+                            {
+                                // You cannot decline the local server. Force approved to true
+                                if (!approved)
+                                {
+                                    if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
+                                    {
+                                        NetworkLog.LogWarning(
+                                            "You cannot decline the host connection. The connection was automatically approved.");
+                                    }
+                                }
+
+                                HandleApproval(ServerClientId, createPlayerObject, playerPrefabHash, true, position, rotation);
+                            });
+                    }
+                    else
+                    {
+                        HandleApproval(ServerClientId, NetworkConfig.PlayerPrefab != null, null, true, null, null);
+                    }
+                }
+
+                if (IsServer)
+                {
+                    SpawnManager.ServerSpawnSceneObjectsOnStartSweep();
+                    OnServerStarted?.Invoke();
+                }
+            }
+
+            return true;
+        }
+
+        public void AddNetworkPrefab(GameObject prefab)
+        {
+            if (State != NetworkManagerState.Inactive)
+            {
+                throw new Exception($"All prefabs must be registered before starting {nameof(NetworkManager)}");
+            }
+
+            NetworkPrefab.VerifyValidPrefab(prefab);
+            NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = prefab });
+        }
+
+#if NETCODE_USE_ADDRESSABLES
+        public void AddNetworkPrefab(AssetReferenceGameObject addressableAsset)
+        {
+            if (State != NetworkManagerState.Inactive)
+            {
+                throw new Exception($"All prefabs must be registered before starting {nameof(NetworkManager)}");
+            }
+
+            NetworkConfig.NetworkAddressables.Add(new NetworkAddressable { Addressable = addressableAsset });
+        }
+        public void AddNetworkPrefab(string address)
```

Comment:
I think we should not have this method at all because it just wraps the ctor and it's confusing.